extensions/binding/org.eclipse.smarthome.binding.astro/src/main/java/org/eclipse/smarthome/binding/astro/handler/AstroThingHandler.java

```diff
@@ -164,112 +184,120 @@ private void restartJobs() {
-                                    .usingJobData(jobDataMap).build();
-                            quartzScheduler.scheduleJob(jobDetail, trigger);
-
-                            if (isPositionalChannelLinked()) {
-                                // positional intervalJob
-                                jobDataMap = new JobDataMap();
-                                jobDataMap.put(AbstractBaseJob.KEY_THING_UID, thingUid);
-                                jobDataMap.put(AbstractBaseJob.KEY_JOB_NAME, "job-positional");
-
-                                Date start = new Date(System.currentTimeMillis() + (thingConfig.getInterval()) * 1000);
-                                trigger = newTrigger().withIdentity("trigger-positional", thingUid).startAt(start)
-                                        .withSchedule(simpleSchedule().repeatForever()
-                                                .withIntervalInSeconds(thingConfig.getInterval()))
-                                        .build();
-                                jobDetail = newJob(PositionalJob.class).withIdentity("job-positional", thingUid)
-                                        .usingJobData(jobDataMap).build();
-                                quartzScheduler.scheduleJob(jobDetail, trigger);
-                                logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
-                                        thingConfig.getInterval(), thingUid);
-                            }
+                    // Startup Job
+                    Thread thread = new Thread(dailyJob);
+                    thread.start();
+                    logger.info("Scheduled astro job-startup-{} for thing {}", typeId, thingUID);
+
+                    // Repeat Scheduled Job every configured seconds
+                    LocalDateTime currentTime = LocalDateTime.now();
+                    LocalDateTime futureTimeWithInterval = currentTime.plusSeconds(thingConfig.getInterval());
+                    Date start = Date.from(futureTimeWithInterval.atZone(ZoneId.systemDefault()).toInstant());
+                    if (isPositionalChannelLinked()) {
+                        Expression expression = new CronExpression(
+                                createCronForRepeatEverySeconds(thingConfig.getInterval()), start);
+                        Job positionalJob = new PositionalJob(thingUID);
+                        if (addJobToQueue(positionalJob)) {
+                            scheduledExecutor.schedule(positionalJob, expression);
+                            logger.info("Scheduled astro job-positional with interval of {} seconds for thing {}",
+                                    thingConfig.getInterval(), thingUID);
                         }
+                        // Execute Positional Job immediately
+                        thread = new Thread(positionalJob);
```

Comment:
Where is this thread cleaned up? Does it stay there forever? For each `restartJobs()` we will create 2 new `Thread`s... That's why I though we could call the `run` method directly since it doesn't contain blocking calls that would block the current `Thread`.